content/docs/extensions/neon-utils.md

```diff
@@ -0,0 +1,116 @@
+---
+title: The neon_utils extension
+subtitle: Monitor how Neon's Autoscaling feature allocates compute resources
+enableTableOfContents: true
+---
+
+The `neon_utils` extension provides a `num_cpus()` function you can use to monitor how Neon's _Autoscaling_ feature allocates vCPU in response to workload. The function returns the current number of allocated vCPUs.
+
+For information about Neon's _Autoscaling_ feature, see [Autoscaling](https://neon.tech/docs/introduction/autoscaling).
+
+## Install the `neon_utils` extension
+
+Install the `neon_utils` extension by running the following `CREATE EXTENSION` statement in the Neon **SQL Editor** or from a client such as `psql` that is connected to Neon.
+
+```sql
+CREATE EXTENSION neon_utils;
+```
+
+For information about using the Neon **SQL Editor**, see [Query with Neon's SQL Editor](/docs/get-started-with-neon/query-with-neon-sql-editor). For information about using the `psql` client with Neon, see [Connect with psql](/docs/connect/query-with-psql-editor).
+
+## Use the `num_cpus()` function
+
+In Neon, computing capacity is measured in _Compute Units (CU)_. One CU has 1 vCPU and 4 GB of RAM, 2 CUs have 2 vCPUs and 8 GB of RAM, and so on. The amount of RAM in GBs is always 4 times the number of CUs. A Neon compute can have anywhere from .25 to 7 CUs.
+
+When you enable Autoscaling for a compute endpoint, you define a minimum and maximum compute size, as shown below:
+
+![Edit compute endpoint dialog showing an Autoscaling configuration](/docs/extensions/edit_compute_endpoint.png)
+
+As your workload changes, computing capacity scales dynamically between the minimum and maximum settings defined in your _Autoscaling_ configuration. To retrieve the number of allocated vCPUs at any point in time, you can run the following query:
+
+```sql
+SELECT num_cpus();
+```
+
+For _Autoscaling_ configuration instructions, see [Compute size and Autoscaling configuration](/docs/manage/endpoints#compute-size-and-autoscaling-configuration).
+
+## Limitations
+
+The following limitations apply:
+
+- The `num_cpus()` function does not return fractional vCPU sizes. If the current number of allocated vCPUs is `.25` or `.5`, the `num_cpus()` function returns `1`.
```

Comment:
This is sad but accurate. Also fyi: autoscaling will do fractional CU > 1, e.g. 3.75 CU. So it's not _just_ .25 and .5 that the function is inaccurate; it always returns the number of CPUs, rounded up to the next whole number.

Reply:
Thank you. I will mention that.